### DISCUSSION TITLE
Fixed Kiwi's bug

### DISCUSSION THREAD

**Discussion by phsreis** [redphx/better-xcloud#388] [General]
With no intention I fixed the bug here. It's not the Better-Xcloud itself. It's the tampermankey extension. I installed kiwi last version, installed tampermankey and Better-Xcloud normally then I deleted tampermankey extension and boom, better Xcloud worked again.

**Comment by KingNothing81** [comment]
But with Kiwi Browser the "Better XCloud" script needs Tampermonkey. If you delete Tampermonkey, how can "Better xCloud" still work? 

Apart from that there is a "Kiwi v123" user agent profile now. If you set this, you don't have to delete anything to make things work.

**Reply by phsreis** [reply]
> But with Kiwi Browser the "Better XCloud" script needs Tampermonkey. If you delete Tampermonkey, how can "Better xCloud" still work? 
> 
> Apart from that there is a "Kiwi v123" user agent profile now. If you set this, you don't have to delete anything to make things work.

I'm really thinking that's a bug.. a lucky bug😂.. I didn't get it too. I'll try to do it again.. I'll delete everything and redo what I did. If it works again I'll send the tutorial.

**Reply by phsreis** [reply]
Here.. I just did a video about it. First I said I deleted it, and it was true, but now I tried to play and as you said. It's not possible to xbetter to work without it. Só I reinstalled tampermankey and worked again.
[https://youtu.be/XoO-IofbRnI?si=ebYqSyNROF-QbGm2]()

**Reply by KingNothing81** [reply]
In your video I can't see if you are using the "Kiwi v123" UA profile now or if it's also working without it for you.

Because the problem with current Kiwi 124.x is that it identifies as being based on Chrome v124, although it's still based on a mixture of v122/123 (according to Kiwi's developer). So it's spoofing a 124.x version number and that's what caused problems. So @redphx added a "Kiwi v123" UA profile, that will make Kiwi identify as a 123.x Chrome based browser. And so it works again with the script.

But Tampermonkey injects the script's code into the xCloud website. So I don't know how this worked for you altough you deleted it. Maybe because the browser was still running. So the script was still active in your current browser session. So it still worked until you closed the browser.

But did you try out the Android app instead of Kiwi? Does it not work as good as using Kiwi for you? What I did is to download and install Chrome and set it as the WebView implementation in the Android app. This gives me a slightly better performance than using the default "Android System WebView". And I don't need the spoofing Kiwi browser anymore.

https://github.com/redphx/better-xcloud-android/wiki/Custom-WebView-implementation

**Reply by phsreis** [reply]
I honestly don't know what happened here either. I tried the app, at first I didn't have a great performance. It was lagging too much, but it seems there was an update. I'll try it again and I'll try the WebView implementation. Until now the best performance I had was using kiwi thats why I'm really trying to save it. About the Kiwi v123 UA profile, I don't know. I just downloaded it from Playstore, then downloaded tampermonkey from it's website and It worked. Hope don't crash soon hahahah
If I really discover something about it, I comunicate here. Thx for the tips btw.
